app/backend/src/tests/test_communities.py

```diff
@@ -869,6 +881,51 @@ def test_LeaveCommunity_regression(db):
         assert not api.GetCommunity(communities_pb2.GetCommunityReq(community_id=c2_id)).member
 
 
+def test_enforce_community_memberships_for_user(testing_communities):
+    """
+    Make sure the user is added to the right communities on signup
+    """
+    with auth_api_session() as (auth_api, metadata_interceptor):
+        res = auth_api.SignupFlow(
+            auth_pb2.SignupFlowReq(
+                basic=auth_pb2.SignupBasic(name="testing", email="email@couchers.org.invalid"),
+                account=auth_pb2.SignupAccount(
+                    username="frodo",
+                    birthdate="1970-01-01",
+                    gender="Bot",
+                    hosting_status=api_pb2.HOSTING_STATUS_CAN_HOST,
+                    city="Country 1, Region 1, City 2",
+                    # == create_coordinate(8)
+                    lat=8,
+                    lng=1,
+                    radius=500,
+                    accept_tos=True,
+                ),
+                feedback=auth_pb2.ContributorForm(),
+            )
+        )
+    with session_scope() as session:
+        email_token = (
+            session.execute(select(SignupFlow).where(SignupFlow.flow_token == res.flow_token)).scalar_one().email_token
+        )
+    with auth_api_session() as (auth_api, metadata_interceptor):
```

Comment:
parentheses necessary?

Reply:
Yep, otherwise it's expecting a second `with` block